test/core/end2end/goaway_server_test.cc

```diff
@@ -81,137 +76,209 @@ static void set_resolve_port(int port) {
-
- private:
-  void MakeDNSRequest(
-      std::function<void(absl::StatusOr<std::vector<grpc_resolved_address>>)>
-          on_done) {
     gpr_mu_lock(&g_mu);
     if (g_resolve_port < 0) {
       gpr_mu_unlock(&g_mu);
-      new grpc_core::DNSCallbackExecCtxScheduler(
-          std::move(on_done), absl::UnknownError("Forced Failure"));
+      engine_->Run([on_resolve = std::move(on_resolve)]() mutable {
+        on_resolve(absl::UnknownError("Forced Failure"));
+      });
     } else {
-      std::vector<grpc_resolved_address> addrs;
-      grpc_resolved_address addr;
-      grpc_sockaddr_in* sa = reinterpret_cast<grpc_sockaddr_in*>(&addr);
-      sa->sin_family = GRPC_AF_INET;
-      sa->sin_addr.s_addr = 0x100007f;
-      sa->sin_port = grpc_htons(static_cast<uint16_t>(g_resolve_port));
-      addr.len = static_cast<socklen_t>(sizeof(*sa));
-      addrs.push_back(addr);
+      std::vector<EventEngine::ResolvedAddress> addrs;
+      struct sockaddr_in in;
+      memset(&in, 0, sizeof(struct sockaddr_in));
+      in.sin_family = GRPC_AF_INET;
+      in.sin_addr.s_addr = 0x100007f;
+      in.sin_port = grpc_htons(static_cast<uint16_t>(g_resolve_port));
+      addrs.emplace_back(reinterpret_cast<const sockaddr*>(&in),
+                         sizeof(struct sockaddr_in));
       gpr_mu_unlock(&g_mu);
-      new grpc_core::DNSCallbackExecCtxScheduler(std::move(on_done),
-                                                 std::move(addrs));
+      engine_->Run([on_resolve = std::move(on_resolve),
+                    addrs = std::move(addrs)]() mutable {
+        on_resolve(std::move(addrs));
+      });
     }
   }
-  std::shared_ptr<grpc_core::DNSResolver> default_resolver_;
-  std::shared_ptr<grpc_event_engine::experimental::EventEngine> engine_;
+  void LookupSRV(LookupSRVCallback on_resolve,
+                 absl::string_view name) override {
+    CHECK(default_resolver_.ok());
+    return (*default_resolver_)->LookupSRV(std::move(on_resolve), name);
+  }
+  void LookupTXT(LookupTXTCallback on_resolve,
+                 absl::string_view name) override {
+    CHECK(default_resolver_.ok());
+    return (*default_resolver_)->LookupTXT(std::move(on_resolve), name);
+  }
+
+ private:
+  std::shared_ptr<EventEngine> engine_;
+  absl::StatusOr<std::unique_ptr<EventEngine::DNSResolver>> default_resolver_;
 };
 
-}  // namespace
+#ifndef GPR_WINDOWS
 
-static grpc_ares_request* my_dns_lookup_ares(
-    const char* dns_server, const char* addr, const char* default_port,
-    grpc_pollset_set* interested_parties, grpc_closure* on_done,
-    std::unique_ptr<grpc_core::EndpointAddressesList>* addresses,
-    int query_timeout_ms) {
-  if (0 != strcmp(addr, "test")) {
-    // A records should suffice
-    return iomgr_dns_lookup_ares(dns_server, addr, default_port,
-                                 interested_parties, on_done, addresses,
-                                 query_timeout_ms);
+class TestEventEngine : public EventEngine {
```

Comment:
This is nearly identical to [examples/cpp/default_event_engine/wrapping_event_engine.h](https://github.com/grpc/grpc/blob/00df46f77d4d79bc45da07916d141d4e2d50f652/examples/cpp/default_event_engine/wrapping_event_engine.h). Let's copy that to [test/core/event_engine/util](https://github.com/grpc/grpc/tree/00df46f77d4d79bc45da07916d141d4e2d50f652/test/core/event_engine/util), and add a constructor that takes a shared_ptr<EventEngine>.

Reply:
Thanks! Merged in https://github.com/grpc/grpc/pull/39050/commits/adb2cf7f98cf7b5413dffa091e344cb376d9e047.